org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/guice/AbstractGuiceAwareExecutableExtensionFactory.java

```diff
@@ -49,15 +50,35 @@ public Object create() throws CoreException {
 		try {
 			final Class<?> clazz = getBundle().loadClass(clazzName);
 			final Injector injector = getInjector();
+			if (injector == null) {
+				throw handleCreationError(null);
+			}
 			final Object result = injector.getInstance(clazz);
 			if (result instanceof IExecutableExtension)
 				((IExecutableExtension) result).setInitializationData(config, null, null);
 			return result;
+		} catch (Throwable e) {
+			throw handleCreationError(e);
 		}
-		catch (Exception e) {
-			log.error(e.getMessage(), e);
-			throw new CoreException(new Status(IStatus.ERROR, getBundle().getSymbolicName(), e.getMessage() + " ExtensionFactory: "+ getClass().getName(), e));
+	}
+	
+	private CoreException handleCreationError (Throwable cause) {
+		Bundle bundle = null;
+		String bundleName = null;
+		try {
+			bundle = getBundle();
+		} catch (NullPointerException e) {
+			// subclasses generated with Xtext <2.13 will raise a NPE when the UI bundle wqs not activated
+			// try to guess the bundle from the EEF's package
+			bundle = Platform.getBundle(getClass().getPackage().getName());
+		}
+		bundleName = bundle != null ? bundle.getSymbolicName() : "UNKNOWN";
+		
+		if (bundle != null && bundle.getState()!=Bundle.ACTIVE) {
+			return new CoreException(new Status(IStatus.ERROR, bundleName, "Bundle "+ bundleName+" not activated. Please check that all dependencies could be resolved and 'Bundle-ActivationPolicy: lazy' is configured in the bundle's MANIFEST.MF. Check also the error log."));	
```

Comment:
And just in case you change anything in the file, could you please fix the formatting?
Here and a line above.